### DISCUSSION TITLE
[Guide] The easiest way to move plausible instance to new VPS

### DISCUSSION THREAD

**Discussion by tvaril** [plausible/analytics#4389] [Self-Hosted Support]
Hi all,

I decided to sum up all I did to move my existing plausible instance to new VPS. It turned out to be the easiest way comparing to others I found around the internet. It took me less then 10 minutes to migrate all and getting it up and running, but I do not have many data there. 

I recently moved to Hetzner's cloud arm64 VPS, which is blazing fast, cheap a powerful comparing to my previous amd64 hosting platforms. Even the plausible website is much more responsive then it was before. I could notice it when browsing stats. It took me multiple tries before I got it working so it is is crucial to do it exactly in this order :-)

**Requirements**
new VPS ready with docker installed

**Detailed step-by-step**

**1. Clone plausible dir on new VPS**  
Clone the dir with clickhouse folders etc and most importantly `docker-compose.yml` and `plausible-conf.env`  from old VPS to the new one using rsync. This is the command:
```bash
sudo rsync -azvP -e 'ssh -p {port number} -i /home/user/.ssh/root-private-key' 'root@[IPv6 address]':/srv/plausible /srv/
```

**2. Get it all up and running on new VPS**  
Enter the cloned directory `cd /srv/plausible` and run `sudo docker compose up -d`, wait until all the images are downloaded and containers started, running healthy etc.

**3. Stop all containers on new VPS** 
Now stop all containers:
 ```bash
sudo docker compose stop plausible plausible_db-1 plausible_events_db-1
 ```

**4. Delete all volumes on new VPS** 
Delete all the volumes which were created in the step 2 (few minutes ago) `sudo docker volume ls`
 ```bash
DRIVER    VOLUME NAME
local     plausible_db-data
local     plausible_event-data
local     plausible_event-logs
 ```
`sudo docker volume rm plausible_db-data plausible_event-data plausible_event-logs`

**5. Stop all containers on old VPS** 
Now move to old VPS and stop all the containers:
 ```bash
sudo docker compose stop plausible plausible_db-1 plausible_events_db-1
 ```

**6. Clone volumes from old VPS to new VPS** 
Move to the new VPS and check your docker volumes directory. I chose docker to use `/srv/docker/`, yours can be the default `/var/lib/docker/`, so adjust the URLs and clone volumes from the old VPS, again using rsync:
```bash
sudo rsync -azvP -e 'ssh -p {port number} -i /home/user/.ssh/root-private-key' 'root@[IPv6 address]':/srv/docker/volumes/plausible_db-data /srv/docker/volumes/
```
```bash
sudo rsync -azvP -e 'ssh -p {port number} -i /home/user/.ssh/root-private-key' 'root@[IPv6 address]':/srv/docker/volumes/plausible_event-data /srv/docker/volumes/
```
```bash
sudo rsync -azvP -e 'ssh -p {port number} -i /home/user/.ssh/root-private-key' 'root@[IPv6 address]':/srv/docker/volumes/plausible_event-logs /srv/docker/volumes/
```

**7. Get it all up and running again on new VPS**  
Enter the cloned directory `cd /srv/plausible` and run `sudo docker compose up -d`, wait until all the containers are running.

**8. Fix the webserver config and DNS** 
Depending on the setup you have, you have to change your webserver config and DNS settings. For myself it was only to clone Caddyfile and to replace CNAME target value with new VPS URL.

**9. Test everything before deleting any data on old VPS**
Make yourself absolutely 💯％ sure that all works as expected before you delete the data on **old VPS**!!!

That's it, hope it helps someone, let me know :-)

Chers

**Comment by ruslandoga** [comment]
👋 @myazaki 

Thank you for the guide!

---

I would warn however to not delete the volumes (step 4) that contain the events data until you are *completely, absolutely, 💯* sure everything was migrated successfully.

**Reply by tvaril** [reply]
You are deleting volumes on **new VPS** (the one you are about to transfer all the data to), created by install script few minutes ago so there shouldn't be anything important there. Anyway, thank you for comment, I still did some tweaking to make it hopefully more clear :-)

**Comment by visini** [comment]
For posterity: I recently wrote a blog post about this topic (I use another approach – I don't copy the volumes, but dumps of the Postgres / ClickHouse tables). See here: https://camillovisini.com/coding/migrating-plausible-analytics

Hope this is helpful for somebody who comes across this in the future 🙏

**Reply by tvaril** [reply]
That is why I called it easiest ;-)

**Comment by js32** [comment]
nice, thx! Bit of a hassle, coming from a previous version of plausible, but worked out in the end :) 
my recommendation, if you're coming from an (upgraded) old installation:
clone plausible repo from GitHub adjust settings in new compose and env file and rename your copied volumes.

**Comment by qlereboursFR** [comment]
For posterity too, I added another guide here: https://github.com/plausible/community-edition/issues/257
where I mention that there's currently an issue when running the migration tool that helps migrating ClickHouse data from v1 to v2. I also mention a solution, even though it's not very clean, I don't see another way to do it at the moment

**Comment by dispatchrabbi** [comment]
Just popping in to say thank you for this guide! It's almost exactly what I needed and it's clear enough that I could adapt it for my needs.
